celsos_bookSearch/celsos_bookSearch/View/SearchViewController.swift

```diff
@@ -9,24 +9,64 @@ import UIKit
 
 class SearchViewController: UIViewController {
 
+    // MARK: - Outlets
+    @IBOutlet weak var searchBarOutlet: UISearchBar!
+    @IBOutlet weak var recentSearchesTableViewOutlet: UITableView!
+
+    var searchesArray: [String] = []
+    let searchCellNibName = "SearchesTableViewCell"
+    let searchCellIdentifier = SearchesTableViewCell().searchCellIdentifier
+
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        self.view.backgroundColor = .green
-        self.title = "Teste Nav"
+        self.navigationController?.setNavigationBarHidden(true, animated: false)
+
+        let tap = UITapGestureRecognizer(target: self, action: #selector(UIInputViewController.dismissKeyboard))
+        view.addGestureRecognizer(tap)
+
+        searchBarOutlet.delegate = self
 
-        // Do any additional setup after loading the view.
+        recentSearchesTableViewOutlet.delegate = self
+        recentSearchesTableViewOutlet.dataSource = self
+        recentSearchesTableViewOutlet.reloadData()
+
+        let searchCellNib = UINib(nibName: searchCellNibName, bundle: Bundle(for: SearchesTableViewCell.self))
+        self.recentSearchesTableViewOutlet.register(searchCellNib, forCellReuseIdentifier:                        self.searchCellIdentifier)
     }
 
+    @objc func dismissKeyboard() {
+        view.endEditing(true)
+    }
 
-    /*
-    // MARK: - Navigation
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        if searchBarOutlet.text != "" {
```

Comment:
That was a nice hint.
But i need to pass the string to the array only when _searchBarOutlet.text_ where not "". So i built this:
`if let searchBarText = searchBarOutlet.text, !searchBarText.isEmpty {...}`
But i think about it again, and it will be good practice if i only add different searches at the array. So i ended up with this little Frankenstein:
`if let searchBarText = searchBarOutlet.text, !searchBarText.isEmpty, !searchesArray.contains(searchBarText) {...}`
What do you think?

Reply:
Nice solution!